dts/bindings/power/state.yaml

```
@@ -0,0 +1,26 @@
+# Copyright (c) 2020, Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Properties for power management state
+
+compatible: "pm-state"
+
+properties:
+    pm-state:
+        type: int
+        required: true
+        description: indicates a power state
+        enum:
+            - 0 # PM_STATE_ACTIVE
+            - 1 # PM_STATE_RUNTIME_IDLE
+            - 2 # PM_STATE_SUSPEND_TO_IDLE
+            - 3 # PM_STATE_STANDBY
+            - 4 # PM_STATE_SUSPEND_TO_RAM
+            - 5 # PM_STATE_SUSPEND_TO_DISK
+            - 6 # PM_STATE_SOFT_OFF
```

enum could use strings, which would be clearer for the reader:
```suggestion
        enum:
            - active
            - runtime_idle
            - suspend_to_idle
            ...
```

---

I'm inclined to agree these should be symbolic.  We need to kickstart #21273.

---

This actually works right now already (https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/bindings/pinctrl/st%2Cstm32-pinctrl.yaml#L76).
Would we need something more for this case?

---

We need to make sure that the values correspond to the constants defined in the C header that are used in runtime API.  I believe the devicetree ordinals are based on the order of the text in the enum sequence, so if one or the other changes value/order things will go wrong in very subtle ways.

---

yep, we can if the values are previously defined right ? we can't just set strings here, right ?

---

that is the problem I brought in the last meeting. I agree with @erwango strings are clearer, but I really hate having this information duplicated in C domain and here. This is very fragile ... The lucky here is that this should not change very often.

Will change to use string while we don't have a better option.

---

I agree that using strings looks better.

Perhaps we should generate DT macros automatically based on enum strings as a new feature in DT macros generator, like:
```
#define DT_COMPAT_PM_STATE_ACTIVE    0
#define DT_COMPAT_PM_STATE_RUNTIME_IDLE    1
...
```
then access them with something like:
```
DT_COMPAT_ENUM_IDX(pm_state, active) -> 0
DT_COMPAT_ENUM_IDX(pm_state, runtime_idle) -> 1
```